Unable to highlight file

sonar-kotlin: 1.0.0 and 0.6.0
sonar: 7.9
maven: 3.6

Build fails with the error:
`Unable to highlight file <<FILE>> Start pointer [line=53, lineOffset=67] should be before end pointer [line=53, lineOffset=67]`

Could not find anything unusual on the pointed location.

Here it is:
```
Caused by: org.apache.maven.plugin.MojoExecutionException: Unable to highlight file <<FILES>>
    at org.sonarsource.scanner.maven.bootstrap.ScannerBootstrapper.execute (ScannerBootstrapper.java:67)
    at org.sonarsource.scanner.maven.SonarQubeMojo.execute (SonarQubeMojo.java:104)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
...
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.IllegalArgumentException: Unable to highlight file <<FILE>>
    at org.sonar.api.batch.sensor.highlighting.internal.DefaultHighlighting.highlight (DefaultHighlighting.java:85)
    at org.sonar.api.batch.sensor.highlighting.internal.DefaultHighlighting.highlight (DefaultHighlighting.java:37)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.highlightByType (KotlinSyntax.kt:41)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.access$highlightByType (KotlinSyntax.kt:13)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax$processFile$1.invoke (KotlinSyntax.kt:27)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax$processFile$1.invoke (KotlinSyntax.kt:13)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens (KotlinSyntax.kt:35)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens (KotlinSyntax.kt:36)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens (KotlinSyntax.kt:36)
...
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens (KotlinSyntax.kt:36)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens (KotlinSyntax.kt:36)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.processFile (KotlinSyntax.kt:18)
    at io.gitlab.arturbosch.detekt.sonar.sensor.FileProcessor.run (FileProcessor.kt:23)
    at io.gitlab.arturbosch.detekt.sonar.sensor.DetektSensor.execute (DetektSensor.kt:27)
    at org.sonar.scanner.sensor.AbstractSensorWrapper.analyse (AbstractSensorWrapper.java:48)
    at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute (ModuleSensorsExecutor.java:85)
    at org.sonar.scanner.sensor.ModuleSensorsExecutor.lambda$execute$1 (ModuleSensorsExecutor.java:59)
    at org.sonar.scanner.sensor.ModuleSensorsExecutor.withModuleStrategy (ModuleSensorsExecutor.java:77)
    at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute (ModuleSensorsExecutor.java:59)
    at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart (ModuleScanContainer.java:82)
    at org.sonar.core.platform.ComponentContainer.startComponents (ComponentContainer.java:136)
    at org.sonar.core.platform.ComponentContainer.execute (ComponentContainer.java:122)
    at org.sonar.scanner.scan.ProjectScanContainer.scan (ProjectScanContainer.java:400)
    at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively (ProjectScanContainer.java:395)
    at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively (ProjectScanContainer.java:392)
    at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart (ProjectScanContainer.java:358)
    at org.sonar.core.platform.ComponentContainer.startComponents (ComponentContainer.java:136)
    at org.sonar.core.platform.ComponentContainer.execute (ComponentContainer.java:122)
    at org.sonar.scanner.bootstrap.GlobalContainer.doAfterStart (GlobalContainer.java:141)
    at org.sonar.core.platform.ComponentContainer.startComponents (ComponentContainer.java:136)
    at org.sonar.core.platform.ComponentContainer.execute (ComponentContainer.java:122)
    at org.sonar.batch.bootstrapper.Batch.doExecute (Batch.java:73)
    at org.sonar.batch.bootstrapper.Batch.execute (Batch.java:67)
    at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute (BatchIsolatedLauncher.java:46)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke (IsolatedLauncherProxy.java:60)
    at com.sun.proxy.$Proxy25.execute (Unknown Source)
    at org.sonarsource.scanner.api.EmbeddedScanner.doExecute (EmbeddedScanner.java:185)
    at org.sonarsource.scanner.api.EmbeddedScanner.execute (EmbeddedScanner.java:137)
    at org.sonarsource.scanner.maven.bootstrap.ScannerBootstrapper.execute (ScannerBootstrapper.java:65)
    at org.sonarsource.scanner.maven.SonarQubeMojo.execute (SonarQubeMojo.java:104)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.IllegalArgumentException: Start pointer [line=53, lineOffset=67] should be before end pointer [line=53, lineOffset=67]
    at org.sonar.api.internal.google.common.base.Preconditions.checkArgument (Preconditions.java:145)
    at org.sonar.api.batch.fs.internal.DefaultInputFile.newRangeValidPointers (DefaultInputFile.java:348)
    at org.sonar.api.batch.fs.internal.DefaultInputFile.newRange (DefaultInputFile.java:311)
    at org.sonar.api.batch.sensor.highlighting.internal.DefaultHighlighting.highlight (DefaultHighlighting.java:83)
    at org.sonar.api.batch.sensor.highlighting.internal.DefaultHighlighting.highlight (DefaultHighlighting.java:37)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.highlightByType (KotlinSyntax.kt:41)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.access$highlightByType (KotlinSyntax.kt:13)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax$processFile$1.invoke (KotlinSyntax.kt:27)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax$processFile$1.invoke (KotlinSyntax.kt:13)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens (KotlinSyntax.kt:35)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens (KotlinSyntax.kt:36)
...
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens (KotlinSyntax.kt:36)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens (KotlinSyntax.kt:36)
    at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.processFile (KotlinSyntax.kt:18)
    at io.gitlab.arturbosch.detekt.sonar.sensor.FileProcessor.run (FileProcessor.kt:23)
    at io.gitlab.arturbosch.detekt.sonar.sensor.DetektSensor.execute (DetektSensor.kt:27)
    at org.sonar.scanner.sensor.AbstractSensorWrapper.analyse (AbstractSensorWrapper.java:48)
    at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute (ModuleSensorsExecutor.java:85)
    at org.sonar.scanner.sensor.ModuleSensorsExecutor.lambda$execute$1 (ModuleSensorsExecutor.java:59)
    at org.sonar.scanner.sensor.ModuleSensorsExecutor.withModuleStrategy (ModuleSensorsExecutor.java:77)
    at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute (ModuleSensorsExecutor.java:59)
    at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart (ModuleScanContainer.java:82)
    at org.sonar.core.platform.ComponentContainer.startComponents (ComponentContainer.java:136)
    at org.sonar.core.platform.ComponentContainer.execute (ComponentContainer.java:122)
    at org.sonar.scanner.scan.ProjectScanContainer.scan (ProjectScanContainer.java:400)
    at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively (ProjectScanContainer.java:395)
    at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively (ProjectScanContainer.java:392)
    at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart (ProjectScanContainer.java:358)
    at org.sonar.core.platform.ComponentContainer.startComponents (ComponentContainer.java:136)
    at org.sonar.core.platform.ComponentContainer.execute (ComponentContainer.java:122)
    at org.sonar.scanner.bootstrap.GlobalContainer.doAfterStart (GlobalContainer.java:141)
    at org.sonar.core.platform.ComponentContainer.startComponents (ComponentContainer.java:136)
    at org.sonar.core.platform.ComponentContainer.execute (ComponentContainer.java:122)
    at org.sonar.batch.bootstrapper.Batch.doExecute (Batch.java:73)
    at org.sonar.batch.bootstrapper.Batch.execute (Batch.java:67)
    at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute (BatchIsolatedLauncher.java:46)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke (IsolatedLauncherProxy.java:60)
    at com.sun.proxy.$Proxy25.execute (Unknown Source)
    at org.sonarsource.scanner.api.EmbeddedScanner.doExecute (EmbeddedScanner.java:185)
    at org.sonarsource.scanner.api.EmbeddedScanner.execute (EmbeddedScanner.java:137)
    at org.sonarsource.scanner.maven.bootstrap.ScannerBootstrapper.execute (ScannerBootstrapper.java:65)
    at org.sonarsource.scanner.maven.SonarQubeMojo.execute (SonarQubeMojo.java:104)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
```

Same error here. Running SonarQube 7.9.1.27448, directly from docker, with sonar-kotlin 1.0.0. Sonarqube scanner gradle plugin @ 2.8.

`org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':sonarqube'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.run(EventFiringTaskExecuter.java:51)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:301)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:293)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:175)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:46)
        at org.gradle.execution.taskgraph.LocalTaskInfoExecutor.execute(LocalTaskInfoExecutor.java:42)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:277)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:262)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:135)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:130)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.execute(DefaultTaskPlanExecutor.java:200)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.executeWithWork(DefaultTaskPlanExecutor.java:191)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.run(DefaultTaskPlanExecutor.java:130)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: java.lang.IllegalArgumentException: Unable to highlight file src/main/kotlin/cz/humatch/rules/MultiplyPolynomRule.kt
        at org.sonar.api.batch.sensor.highlighting.internal.DefaultHighlighting.highlight(DefaultHighlighting.java:85)
        at org.sonar.api.batch.sensor.highlighting.internal.DefaultHighlighting.highlight(DefaultHighlighting.java:37)
        at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.highlightByType(KotlinSyntax.kt:41)
        at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.access$highlightByType(KotlinSyntax.kt:13)
        at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax$processFile$1.invoke(KotlinSyntax.kt:20)
        at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax$processFile$1.invoke(KotlinSyntax.kt:13)
        at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens(KotlinSyntax.kt:35)
        at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens(KotlinSyntax.kt:36)
        at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens(KotlinSyntax.kt:36)
        at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens(KotlinSyntax.kt:36)
        at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens(KotlinSyntax.kt:36)
        at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens(KotlinSyntax.kt:36)
        at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.visitTokens(KotlinSyntax.kt:36)
        at io.gitlab.arturbosch.detekt.sonar.foundation.KotlinSyntax.processFile(KotlinSyntax.kt:18)
        at io.gitlab.arturbosch.detekt.sonar.sensor.FileProcessor.run(FileProcessor.kt:23)
        at io.gitlab.arturbosch.detekt.sonar.sensor.DetektSensor.execute(DetektSensor.kt:27)
        at org.sonar.scanner.sensor.AbstractSensorWrapper.analyse(AbstractSensorWrapper.java:48)
        at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute(ModuleSensorsExecutor.java:85)
        at org.sonar.scanner.sensor.ModuleSensorsExecutor.lambda$execute$1(ModuleSensorsExecutor.java:59)
        at org.sonar.scanner.sensor.ModuleSensorsExecutor.withModuleStrategy(ModuleSensorsExecutor.java:77)
        at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute(ModuleSensorsExecutor.java:59)
        at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:82)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
        at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:400)
        at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:395)
        at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:358)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
        at org.sonar.scanner.bootstrap.GlobalContainer.doAfterStart(GlobalContainer.java:141)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
        at org.sonar.batch.bootstrapper.Batch.doExecute(Batch.java:73)
        at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:67)
        at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:46)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
        at com.sun.proxy.$Proxy895.execute(Unknown Source)
        at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:189)
        at org.sonarsource.scanner.api.EmbeddedScanner.execute(EmbeddedScanner.java:138)
        at org.sonarqube.gradle.SonarQubeTask.run(SonarQubeTask.java:100)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:801)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:768)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:301)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:293)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:175)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
        ... 31 more
Caused by: java.lang.IllegalArgumentException: 1579 is not a valid offset for file src/main/kotlin/cz/humatch/rules/MultiplyPolynomRule.kt. Max offset is 1576
        at org.sonar.api.internal.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
        at org.sonar.api.batch.fs.internal.DefaultInputFile.newPointer(DefaultInputFile.java:317)
        at org.sonar.api.batch.fs.internal.DefaultInputFile.newRange(DefaultInputFile.java:311)
        at org.sonar.api.batch.sensor.highlighting.internal.DefaultHighlighting.highlight(DefaultHighlighting.java:83)
        ... 90 more`

The file in question is attached. If I remove some lines at random, it works, but fails on another file. Please advise, I would very much like to use sonar-kotlin and not the built-in kotlin plugin. Please find the relevant file attached.

[MultiplyPolynomRule.kt.txt](https://github.com/arturbosch/sonar-kotlin/files/3691733/MultiplyPolynomRule.kt.txt)

--- a/pom.xml
+++ b/pom.xml
@@ -5,17 +5,18 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>io.gitlab.arturbosch</groupId>
     <artifactId>sonar-kotlin</artifactId>
-    <version>1.0.0</version>
+    <version>1.1.1</version>
 
     <packaging>sonar-plugin</packaging>
 
     <properties>
         <kotlin.version>1.3.50</kotlin.version>
         <spek.version>2.0.2</spek.version>
-        <detekt.version>1.0.0</detekt.version>
+        <detekt.version>1.1.1</detekt.version>
         <sonar.api.version>6.7.7</sonar.api.version>
-        <assertj.version>3.12.2</assertj.version>
+        <assertj.version>3.13.2</assertj.version>
         <jcommander.version>1.78</jcommander.version>
+        <mockk.version>1.9.3</mockk.version>
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
         <kotlin.compiler.incremental>true</kotlin.compiler.incremental>
@@ -120,7 +121,7 @@
         <dependency>
             <groupId>io.mockk</groupId>
             <artifactId>mockk</artifactId>
-            <version>1.9.3</version>
+            <version>${mockk.version}</version>
             <scope>test</scope>
         </dependency>
     </dependencies>

--- a/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/foundation/KotlinSyntax.kt
+++ b/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/foundation/KotlinSyntax.kt
@@ -1,30 +1,65 @@
 package io.gitlab.arturbosch.detekt.sonar.foundation
 
 import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.openapi.editor.Document
+import org.jetbrains.kotlin.com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.diagnostics.PsiDiagnosticUtils
 import org.jetbrains.kotlin.lexer.KtTokens
 import org.jetbrains.kotlin.psi.KtAnnotationEntry
 import org.jetbrains.kotlin.psi.KtFile
 import org.jetbrains.kotlin.psi.psiUtil.getNonStrictParentOfType
 import org.sonar.api.batch.fs.InputFile
 import org.sonar.api.batch.sensor.SensorContext
-import org.sonar.api.batch.sensor.highlighting.NewHighlighting
 import org.sonar.api.batch.sensor.highlighting.TypeOfText
 
 object KotlinSyntax {
 
     fun processFile(inputFile: InputFile, ktFile: KtFile, context: SensorContext) {
         val syntax = context.newHighlighting().onFile(inputFile)
+        val document: Document = ktFile.viewProvider.document ?: return
+
+        fun positions(psi: PsiElement) =
+            PsiDiagnosticUtils.offsetToLineAndColumn(document, psi.textRange.startOffset) to
+                PsiDiagnosticUtils.offsetToLineAndColumn(document, psi.textRange.endOffset)
+
+        fun highlightByType(psi: PsiElement, type: TypeOfText) {
+            val (start, end) = positions(psi)
+            syntax.highlight(
+                inputFile.newRange(
+                    start.line,
+                    start.column - 1,
+                    end.line,
+                    end.column - 1
+                ),
+                type
+            )
+        }
+
+        fun highlightByType(astNode: ASTNode, type: TypeOfText) {
+            val psi = astNode.psi
+            if (psi != null) {
+                highlightByType(psi, type)
+            }
+        }
+
+        fun handleAnnotation(astNode: ASTNode) {
+            val annotationEntry = astNode.psi
+                ?.getNonStrictParentOfType(KtAnnotationEntry::class.java)
+            if (annotationEntry != null) {
+                highlightByType(annotationEntry, TypeOfText.ANNOTATION)
+            }
+        }
 
         ktFile.node.visitTokens {
             when (it.elementType) {
-                in KtTokens.KEYWORDS -> syntax.highlightByType(it, TypeOfText.KEYWORD)
-                in KtTokens.SOFT_KEYWORDS -> syntax.highlightByType(it, TypeOfText.KEYWORD)
-                in KtTokens.STRINGS -> syntax.highlightByType(it, TypeOfText.STRING)
-                in KtTokens.COMMENTS -> syntax.highlightByType(it, TypeOfText.COMMENT)
-                KtTokens.OPEN_QUOTE, KtTokens.CLOSING_QUOTE -> syntax.highlightByType(it, TypeOfText.STRING)
-                KtTokens.AT -> syntax.handleAnnotations(it)
+                in KtTokens.KEYWORDS -> highlightByType(it, TypeOfText.KEYWORD)
+                in KtTokens.SOFT_KEYWORDS -> highlightByType(it, TypeOfText.KEYWORD)
+                in KtTokens.STRINGS -> highlightByType(it, TypeOfText.STRING)
+                in KtTokens.COMMENTS -> highlightByType(it, TypeOfText.COMMENT)
+                KtTokens.OPEN_QUOTE, KtTokens.CLOSING_QUOTE -> highlightByType(it, TypeOfText.STRING)
+                KtTokens.AT -> handleAnnotation(it)
                 KtTokens.SHORT_TEMPLATE_ENTRY_START, KtTokens.LONG_TEMPLATE_ENTRY_START, KtTokens.ESCAPE_SEQUENCE,
-                KtTokens.LONG_TEMPLATE_ENTRY_END -> syntax.highlightByType(it, TypeOfText.ANNOTATION)
+                KtTokens.LONG_TEMPLATE_ENTRY_END -> highlightByType(it, TypeOfText.ANNOTATION)
             }
         }
 
@@ -34,19 +69,5 @@ object KotlinSyntax {
     private fun ASTNode.visitTokens(currentNode: (node: ASTNode) -> Unit) {
         currentNode(this)
         getChildren(null).forEach { it.visitTokens(currentNode) }
-    }
-
-    private fun NewHighlighting.highlightByType(astNode: ASTNode, type: TypeOfText) {
-        val range = astNode.textRange
-        highlight(range.startOffset, range.endOffset, type)
-    }
-
-    private fun NewHighlighting.handleAnnotations(astNode: ASTNode) {
-        val psi = astNode.psi
-        val annotationEntry = psi.getNonStrictParentOfType(KtAnnotationEntry::class.java)
-        if (annotationEntry != null) {
-            val range = annotationEntry.textRange
-            highlight(range.startOffset, range.endOffset, TypeOfText.ANNOTATION)
-        }
     }
 }

--- a/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/foundation/NoAutoCorrectConfig.kt
+++ b/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/foundation/NoAutoCorrectConfig.kt
@@ -18,5 +18,4 @@ class NoAutoCorrectConfig(private val config: Config) : Config {
     }
 
     override fun <T : Any> valueOrNull(key: String): T? = config.valueOrNull(key)
-
 }

--- a/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/jacoco/KotlinJacocoReportAnalyzer.kt
+++ b/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/jacoco/KotlinJacocoReportAnalyzer.kt
@@ -36,7 +36,6 @@ import org.sonar.plugins.java.api.JavaResourceLocator
 import java.io.File
 import java.util.ArrayList
 
-
 private const val NO_JACOCO_EXECUTION_DATA_MESSAGE =
     "Project coverage is set to 0% as no JaCoCo execution data has been dumped: {}"
 private const val NO_CLASS_FILES_MESSAGE =
@@ -53,7 +52,8 @@ class KotlinJacocoReportAnalyzer @JvmOverloads constructor(
     private val javaResourceLocator: JavaResourceLocator,
     private val kotlinClasspath: KotlinClasspath,
     private val readCoveragePerTests: Boolean = true,
-    private val report: File) {
+    private val report: File
+) {
 
     private var classFilesCache: MutableMap<String, File> = Maps.newHashMap()
 
@@ -68,8 +68,9 @@ class KotlinJacocoReportAnalyzer @JvmOverloads constructor(
         val inputFile = javaResourceLocator.findResourceByClassName(className) ?: return null
         return if (inputFile.type() == InputFile.Type.TEST) {
             null
-        } else inputFile
-
+        } else {
+            inputFile
+        }
     }
 
     fun analyse(context: SensorContext) {
@@ -94,7 +95,7 @@ class KotlinJacocoReportAnalyzer @JvmOverloads constructor(
                 populateClassFilesCache(file, path + file.name + "/")
             } else if (file.name.endsWith(".class")) {
                 val className = path + StringUtils.removeEnd(file.name, ".class")
-                classFilesCache.put(className, file)
+                classFilesCache[className] = file
             }
         }
     }
@@ -133,8 +134,10 @@ class KotlinJacocoReportAnalyzer @JvmOverloads constructor(
         }
     }
 
-    private fun readCoveragePerTests(executionDataVisitor: ExecutionDataVisitor,
-                                     jacocoReportReader: JacocoReportReader): Boolean {
+    private fun readCoveragePerTests(
+        executionDataVisitor: ExecutionDataVisitor,
+        jacocoReportReader: JacocoReportReader
+    ): Boolean {
         var collectedCoveragePerTest = false
         if (readCoveragePerTests) {
             for ((key, value) in executionDataVisitor.sessions) {
@@ -146,9 +149,11 @@ class KotlinJacocoReportAnalyzer @JvmOverloads constructor(
         return collectedCoveragePerTest
     }
 
-    private fun analyzeLinesCoveredByTests(sessionId: String,
-                                           executionDataStore: ExecutionDataStore,
-                                           jacocoReportReader: JacocoReportReader): Boolean {
+    private fun analyzeLinesCoveredByTests(
+        sessionId: String,
+        executionDataStore: ExecutionDataStore,
+        jacocoReportReader: JacocoReportReader
+    ): Boolean {
         val i = sessionId.indexOf(' ')
         if (i < 0) {
             return false
@@ -161,7 +166,7 @@ class KotlinJacocoReportAnalyzer @JvmOverloads constructor(
             val resource = getResource(coverage)
             if (resource != null) {
                 val coveredLines = coveredLines(coverage)
-                if (!coveredLines.isEmpty()) {
+                if (coveredLines.isNotEmpty()) {
                     result = true
                 }
             }
@@ -218,5 +223,4 @@ class KotlinJacocoReportAnalyzer @JvmOverloads constructor(
             lineId++
         }
     }
-
 }

--- a/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/rules/DetektRuleKey.kt
+++ b/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/rules/DetektRuleKey.kt
@@ -34,10 +34,12 @@ val ruleKeys = allLoadedRules.map { defineRuleKey(it) }
 
 val ruleKeyLookup = ruleKeys.map { it.ruleKey to it }.toMap()
 
-data class DetektRuleKey(private val repositoryKey: String,
-                         val ruleKey: String,
-                         val active: Boolean,
-                         val issue: Issue) : RuleKey(repositoryKey, ruleKey) {
+data class DetektRuleKey(
+    private val repositoryKey: String,
+    val ruleKey: String,
+    val active: Boolean,
+    val issue: Issue
+) : RuleKey(repositoryKey, ruleKey) {
     override fun hashCode(): Int {
         return super.hashCode()
     }

--- a/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/rules/DetektRulesDefinition.kt
+++ b/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/rules/DetektRulesDefinition.kt
@@ -15,7 +15,6 @@ class DetektRulesDefinition : RulesDefinition {
             .createRules()
             .done()
     }
-
 }
 
 fun RulesDefinition.NewRepository.createRules() = apply {

--- a/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/sensor/FileMeasurementStorage.kt
+++ b/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/sensor/FileMeasurementStorage.kt
@@ -23,8 +23,12 @@ class FileMeasurementStorage(private val context: SensorContext) {
         save(file, inputComponent, complexityKey, COMPLEXITY)
     }
 
-    private fun save(file: KtFile, inputComponent: InputComponent,
-                     dataKey: Key<Int>, metricKey: Metric<Int>) {
+    private fun save(
+        file: KtFile,
+        inputComponent: InputComponent,
+        dataKey: Key<Int>,
+        metricKey: Metric<Int>
+    ) {
         file.getUserData(dataKey)?.let {
             context.newMeasure<Int>()
                 .withValue(it)

--- a/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/sensor/IssueReporter.kt
+++ b/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/sensor/IssueReporter.kt
@@ -13,8 +13,10 @@ import org.sonar.api.batch.sensor.issue.NewIssue
 import org.sonar.api.config.Configuration
 import java.io.File
 
-class IssueReporter(private val detektion: Detektion,
-                    private val context: SensorContext) {
+class IssueReporter(
+    private val detektion: Detektion,
+    private val context: SensorContext
+) {
 
     private val fileSystem = context.fileSystem()
     private val baseDir = fileSystem.baseDir()

--- a/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/sensor/ProjectMeasurementStorage.kt
+++ b/src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/sensor/ProjectMeasurementStorage.kt
@@ -10,8 +10,10 @@ import org.jetbrains.kotlin.com.intellij.openapi.util.Key
 import org.sonar.api.batch.sensor.SensorContext
 import org.sonar.api.measures.Metric
 
-class ProjectMeasurementStorage(private val detektion: Detektion,
-                                private val context: SensorContext) {
+class ProjectMeasurementStorage(
+    private val detektion: Detektion,
+    private val context: SensorContext
+) {
 
     fun run() {
         save(linesKey, projectLocMetric)

